src/Hive.IO/Hive.IO/Building/Zone.cs

```diff
@@ -684,13 +684,11 @@ private static bool CheckWindowsSelfIntersect(rg.Surface[] windows, double toler
                 {
                     rg.Curve[] intersectionCrvs;
                     rg.Point3d[] intersectionPts;
-                    //misses overlaps ?!
-                    //rg.Intersect.Intersection.SurfaceSurface(windows[i], windows[u], tolerance, out intersectionCrvs, out intersectionPts);
                     rg.Intersect.Intersection.BrepSurface(w1, windows[u], tolerance, out intersectionCrvs, out intersectionPts);
-                    if (intersectionPts.Length > 0)
-                        return false;
                     if (intersectionCrvs.Length > 0)
-                        return false;
+                        foreach(var crv in intersectionCrvs)
+                            if (crv.IsClosed)   // they always have to be closed for intersections. otherwise the windows might just be touching
```

Comment:
idk about the rhino api so i defer to you, but not really sure what this means.  And i see you return the moment you see an intersection which makes sense to me.